pkg/kopia/command/log_output_utils.go

```
@@ -0,0 +1,158 @@
+	}
+	return snapID, backupPath, nil
+}
+
+// SnapshotInfoFromSnapshotCreateOutput returns snapshot ID and root ID from snapshot create output
+func SnapshotInfoFromSnapshotCreateOutput(output string) (string, string, error) {
+	snapID := ""
+	rootID := ""
+	scanner := bufio.NewScanner(strings.NewReader(output))
+	for scanner.Scan() {
+		snapManifest := &snapshot.Manifest{}
+		err := json.Unmarshal([]byte(scanner.Text()), snapManifest)
+		if err != nil {
+			continue
+		}
+		if snapManifest != nil {
+			snapID = string(snapManifest.ID)
+			if snapManifest.RootEntry != nil {
+				rootID = string(snapManifest.RootEntry.ObjectID)
+			}
+		}
+	}
+	if snapID == "" {
+		return "", "", errors.New("Failed to get snapshot ID from create snapshot output")
+	}
+	if rootID == "" {
+		return "", "", errors.New("Failed to get root ID from create snapshot output")
+	}
+	return snapID, rootID, nil
+}
+
+// SnapSizeStatsFromSnapListAll returns a list of snapshot logical sizes assuming the input string
+// is formatted as the output of a kopia snapshot list --all command.
+func SnapSizeStatsFromSnapListAll(output string) (totalSizeB int64, numSnapshots int, err error) {
+	if output == "" {
+		return 0, 0, errors.New("Received empty output")
+	}
+
+	snapList, err := parseSnapshotManifestList(output)
+	if err != nil {
+		return 0, 0, errors.Wrap(err, "Parsing snapshot list output as snapshot manifest list")
+	}
+
+	totalSizeB = sumSnapshotSizes(snapList)
+
+	return totalSizeB, len(snapList), nil
+}
+
+func sumSnapshotSizes(snapList []*snapshot.Manifest) (sum int64) {
+	noSizeDataCount := 0
+	for _, snapInfo := range snapList {
+		if snapInfo.RootEntry == nil ||
+			snapInfo.RootEntry.DirSummary == nil {
+			noSizeDataCount++
+
+			continue
+		}
+
+		sum += snapInfo.RootEntry.DirSummary.TotalFileSize
+	}
+
+	if noSizeDataCount > 0 {
+		log.Error().Print("Found snapshot manifests without size data", field.M{"count": noSizeDataCount})
```

Should we return the error and let the caller handle it?